replay_provider.go

```diff
@@ -1,56 +1,127 @@
 package sse
 
 import (
+	"errors"
+	"strconv"
 	"time"
 )
 
+// NewFiniteReplayProvider creates a finite replay provider with the given max
+// count and auto ID behaviour.
+//
+// Count is the maximum number of events FiniteReplayProvider should hold as
+// valid. It must be greater than zero.
+//
+// AutoIDs configures FiniteReplayProvider to automatically set the IDs of
+// events.
+func NewFiniteReplayProvider(
+	count int, autoIDs bool,
+) (*FiniteReplayProvider, error) {
+	if count < 1 {
```

Comment:
I actually think count should be at least 2. If you think about it, if only one event is buffered, only the event which the client has last received can be replayed. What do you think?

Reply:
True, it would work, but be pointless.